spirv_msl.hpp

```diff
@@ -49,72 +49,70 @@ struct MSLVertexAttr
 	bool per_instance = false;
 	MSLVertexFormat format = MSL_VERTEX_FORMAT_OTHER;
 	spv::BuiltIn builtin = spv::BuiltInMax;
-	bool used_by_shader = false;
 };
 
 // Matches the binding index of a MSL resource for a binding within a descriptor set.
 // Taken together, the stage, desc_set and binding combine to form a reference to a resource
-// descriptor used in a particular shading stage. Generally, only one of the buffer, texture,
-// or sampler elements will be populated. The used_by_shader flag is set to true during
-// compilation of SPIR-V to MSL if the shader makes use of this vertex attribute.
+// descriptor used in a particular shading stage.
 struct MSLResourceBinding
 {
-	spv::ExecutionModel stage;
+	spv::ExecutionModel stage = spv::ExecutionModelMax;
 	uint32_t desc_set = 0;
 	uint32_t binding = 0;
-
-	uint32_t msl_buffer = 0;
```

Comment:
While refactoring, this seemed a bit redundant. @billhollings @cdavis5e, is it needed to have 3 different bindings? Bindings of different types cannot alias in SPIR-V. Are you okay with this change?

Reply:
What about combined image-samplers? Those consume both a texture and a sampler binding.

Reply:
Oh, right, that's a good catch. I'll change it back.